src/system/settings/settings-gamepad.ts

```diff
@@ -99,12 +99,12 @@ export function setSettingGamepad(setting: SettingGamepad, value: number): boole
         if (globalScene.ui && gp) {
           const cancelHandler = () => {
             globalScene.ui.revertMode();
-            (globalScene.ui.getHandler() as GamepadSettingsUiHandler).setOptionCursor(
+            (globalScene.ui.getCurrentHandler() as GamepadSettingsUiHandler).setOptionCursor(
               Object.values(SettingGamepad).indexOf(SettingGamepad.Controller),
               0,
               true,
             );
-            (globalScene.ui.getHandler() as GamepadSettingsUiHandler).updateBindings();
+            (globalScene.ui.getCurrentHandler() as GamepadSettingsUiHandler).updateBindings();
```

Comment:
NIT: Use generic param `<T>`

```suggestion
            globalScene.ui.getCurrentHandler<GamepadSettingsUiHandler>().updateBindings();
```